src/main/java/database/models/FormplayerCaseIndexTable.java

```diff
@@ -0,0 +1,371 @@
+package database.models;
+
+import org.commcare.api.persistence.SqlHelper;
+import org.commcare.api.persistence.SqliteIndexedStorageUtility;
+import org.commcare.cases.model.Case;
+import org.commcare.cases.model.CaseIndex;
+import org.commcare.cases.query.queryset.DualTableSingleMatchModelQuerySet;
+import org.commcare.modern.database.DatabaseHelper;
+import org.commcare.modern.database.DatabaseIndexingUtils;
+import org.commcare.modern.database.TableBuilder;
+import org.commcare.modern.util.Pair;
+import org.sqlite.javax.SQLiteConnectionPoolDataSource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+
+/**
+ * @author ctsims
+ */
+public class FormplayerCaseIndexTable implements org.commcare.modern.engine.cases.CaseIndexTable {
+    public static final String TABLE_NAME = "case_index_storage";
+
+    private static final String COL_CASE_RECORD_ID = "case_rec_id";
+    private static final String COL_INDEX_NAME = "name";
+    private static final String COL_INDEX_TYPE = "type";
+    private static final String COL_INDEX_TARGET = "target";
+
+    SQLiteConnectionPoolDataSource dataSource;
+
+    //TODO: We should do some synchronization to make it the case that nothing can hold
+    //an object for the same cache at once and let us manage the lifecycle
+
+    public FormplayerCaseIndexTable(SQLiteConnectionPoolDataSource dataSource) {
+        this.dataSource = dataSource;
+        Connection connection = null;
+        try {
+            connection = dataSource.getConnection();
+            execSQL(connection, getTableDefinition());
+            createIndexes(connection);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            if (connection != null) {
+                try {
+                    connection.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    private static void execSQL(Connection connection, String query) {
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = connection.prepareStatement(query);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            e.printStackTrace();
```

Comment:
print, would be useful if there was some kind of java data structure that allowed one to make this dryer